### ISSUE TITLE
option to output build version

### ISSUE THREAD

**Issue by ailrst** [UQ-PAC/bil-to-boogie-translator#85]
It would be nice (e.g. for godbolt) to have basil be able to output its own version, ideally at this stage this could just be the commit sha it was built from.

**Comment by l-kent** [comment]
Yeah, I don't think we'd need more than just the commit SHA for now

**Comment by ziggyfish** [comment]
This is generally done through git tags, generally when you create a new tag for a commit. Most build systems (including this one), will give the git version or tag during the build process through environment variables. These environment variables can then be picked up by sbt ( `sys.env.get("NAME_OF_ENV_VAR")` ) and compiled into the package, accessable by ( `version = getClass.getPackage.getImplementationVersion` ) .

**Comment by ailrst** [comment]
The plan is to add an sbt action to write the current git tag to a version file, and have that set to the version in scala that is output from the program. It is possibly also useful to have a 'test build' sbt action which derives a new version from the current git commit.
